app/components/icon_component.rb

```diff
@@ -0,0 +1,14 @@
+class IconComponent < BaseComponent
+  include AssetHelper
+
+  attr_reader :icon, :tag_options
+
+  def initialize(icon:, **tag_options)
+    @icon = icon
+    @tag_options = tag_options
+  end
+
+  def icon_path
+    asset_path("#{design_system_asset_path('img/sprite.svg')}##{icon}")
```

Comment:
(supernit, feel free to ignore)
the `##` is bugging me but I don't think either of these are much better:
```suggestion
    asset_path([design_system_asset_path('img/sprite.svg')}, '#', icon].join)
```
```suggestion
    asset_path(format("%{image}#%{anchor}%", image: design_system_asset_path('img/sprite.svg'), anchor: icon))
```

Reply:
Also just out of curiosity, will this error at all if the icon doesn't exist in the file?

Reply:
The old behavior didn't validate, so it's not like we'd be losing anything by leaving the behavior as-is

We could have like an `ICONS = %w[ a b c ].to_set` constant or something, but 241 icons is a lot and may not be worth our time

Reply:
> (supernit, feel free to ignore)
> the `##` is bugging me but I don't think either of these are much better:

Updated to `string#join` in 31b577d78.

Reply:
> We could have like an `ICONS = %w[ a b c ].to_set` constant or something, but 241 icons is a lot and may not be worth our time

I think it can have a nice added benefit of serving as a developer reference, so I added in acff47184. I wasn't sure about formatting with this (i.e. collapsing some lines), but this way feels easiest to maintain as icons are added/removed in the future.